### PR TITLE
bugfix: TemplateNotFound on windows

### PR DESCRIPTION
**Pull request by JaysonAlbert**
for the template name of the function get_template, "/" is used as the path separator, even on Windows.

original code `os.path.join(self.model, template)` use '\\' as separator which cause an TemplateNotFound Exception

image

```
    @internalcode
    def get_template(
        self,
        name: t.Union[str, "Template"],
        parent: t.Optional[str] = None,
        globals: t.Optional[t.MutableMapping[str, t.Any]] = None,
    ) -> "Template":
        """Load a template by name with :attr:`loader` and return a
        :class:`Template`. If the template does not exist a
        :exc:`TemplateNotFound` exception is raised.

        :param name: Name of the template to load. When loading
            templates from the filesystem,"/" is used as the path
            separator, even on Windows.
```

### stacktrace

```
    |                ^^^^^^^^^^^^^^^^
    |   File "e:\IdeaProjects\skyvern\.venv\Lib\site-packages\fastapi\routing.py", line 301, in app
    |     raw_response = await run_endpoint_function(
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "e:\IdeaProjects\skyvern\.venv\Lib\site-packages\fastapi\routing.py", line 212, in run_endpoint_function
    |     return await dependant.call(**values)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "E:\IdeaProjects\skyvern\skyvern\forge\sdk\routes\agent_protocol.py", line 874, in generate_task
    |     llm_prompt = prompt_engine.load_prompt("generate-task", user_prompt=data.prompt)
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "E:\IdeaProjects\skyvern\skyvern\forge\sdk\prompting.py", line 102, in load_prompt
    |     jinja_template = self.env.get_template(f"{template}.j2")
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "e:\IdeaProjects\skyvern\.venv\Lib\site-packages\jinja2\environment.py", line 1013, in get_template
    |     return self._load_template(name, globals)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "e:\IdeaProjects\skyvern\.venv\Lib\site-packages\jinja2\environment.py", line 972, in _load_template
    |     template = self.loader.load(self, name, self.make_globals(globals))
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "e:\IdeaProjects\skyvern\.venv\Lib\site-packages\jinja2\loaders.py", line 126, in load
    |     source, filename, uptodate = self.get_source(environment, name)
    |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "e:\IdeaProjects\skyvern\.venv\Lib\site-packages\jinja2\loaders.py", line 197, in get_source
    |     pieces = split_template_path(template)
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "e:\IdeaProjects\skyvern\.venv\Lib\site-packages\jinja2\loaders.py", line 36, in split_template_path
    |     raise TemplateNotFound(template)
    | jinja2.exceptions.TemplateNotFound: skyvern\generate-task.j2
```

----

> [!IMPORTANT]
> Fix `TemplateNotFound` error in `load_prompt()` by using forward slashes for path separation in `prompting.py`.
> 
>   - **Bug Fix**:
>     - Fix `TemplateNotFound` exception in `load_prompt()` in `prompting.py` by replacing `os.path.join` with `'/'.join()` to ensure forward slashes are used as path separators, even on Windows.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=Skyvern-AI%2Fskyvern&utm_source=github&utm_medium=referral)<sup> for 32fbdc2b1ce96dc6329c98239e7a798a6a6cb94e. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Review by wintonzheng — APPROVED** [review]
interesting bug. will do a test before merging

**Comment by wintonzheng** [comment]
tested and it works.

@JaysonAlbert curious:
does this line work for you in windows?
https://github.com/Skyvern-AI/skyvern/pull/1185/files#diff-31ca9bf7c60cea7680bd4215aad94f8e9fa1f260268e8ca398f2dc3063296ddaR54

File: skyvern/forge/sdk/prompting.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def load_prompt(self, template: str, **kwargs: Any) -> str:
             str: The populated template.
         """
         try:
-            template = os.path.join(self.model, template)
+            template = '/'.join([self.model, template])
```
Comment (wintonzheng) [inline skyvern/forge/sdk/prompting.py:101]:
```suggestion
            template = "/".join([self.model, template])
```

our linter suggests double quotes

**Comment by JaysonAlbert** [comment]
> tested and it works.
> 
> @JaysonAlbert curious: does this line work for you in windows? https://github.com/Skyvern-AI/skyvern/pull/1185/files#diff-31ca9bf7c60cea7680bd4215aad94f8e9fa1f260268e8ca398f2dc3063296ddaR54
> 
> Can we also update line 54?
> 
> ```
>                 for d in glob.glob(os.path.join(models_dir, "*/"))
> ```

I think it works

**Comment by wintonzheng** [comment]
> > tested and it works.
> > @JaysonAlbert curious: does this line work for you in windows? https://github.com/Skyvern-AI/skyvern/pull/1185/files#diff-31ca9bf7c60cea7680bd4215aad94f8e9fa1f260268e8ca398f2dc3063296ddaR54
> > Can we also update line 54?
> > ```
> >                 for d in glob.glob(os.path.join(models_dir, "*/"))
> > ```
> 
> I think it works

thanks for confirming. merging this commit now

**Pull request merged**